### PR TITLE
Merchant invoices

### LINKED ISSUE #25: Merchant Invoice Show Page: Invoice Item Information
As a merchant
When I visit my merchant invoice show page
Then I see all of my items on the invoice including:
- Item name
- The quantity of the item ordered
- The price the Item sold for
- The Invoice Item status
And I do not see any information related to Items for other merchants

### LINKED ISSUE #26: Merchant Invoice Show Page
As a merchant
When I visit my merchant's invoice show page(/merchants/merchant_id/invoices/invoice_id)
Then I see information related to that invoice including:
- Invoice id
- Invoice status
- Invoice created_at date in the format "Monday, July 18, 2019"
- Customer first and last name

### LINKED ISSUE #27: Merchant Invoices Index
As a merchant,
When I visit my merchant's invoices index (/merchants/merchant_id/invoices)
Then I see all of the invoices that include at least one of my merchant's items
And for each invoice I see its id
And each id links to the merchant invoice show page

### PR DESCRIPTION
**Pull request by Sidarth20**
## Changes
Created feature & model tests for the Merchant Invoices Index, Show Page, & Invoice Item Info

## Checklist
- [x] Tests written where required
- [x] All tests passing
- [x] SimpleCov at 100%

closes #25
closes #26 
closes #27

### CODE REVIEW

File: app/controllers/merchants/invoices_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Merchants::InvoicesController < ApplicationController
+
+  def index
+    @merchant = Merchant.find(params[:merchant_id])
+    @merchant_invoices = @merchant.items.merchant_invoices
```
Comment (mkrumholz) [inline app/controllers/merchants/invoices_controller.rb]: Is this technically a law of demeter violation? That'd be a nitpick, so not a huge concern here, I'd say

File: app/views/merchants/invoices/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <% @invoice_items.each do |invoice_item|  %>
+    <tr>
+      <td> <%= invoice_item.item_name %> </td>
+      <td> <%= invoice_item.quantity %> </td>
+      <td> <%= invoice_item.unit_price %> </td>
+      <td> <%= invoice_item.status %>  </td>
+    </tr>
+    <% end %>
+  </table>
+</section>
```
Comment (mkrumholz) [inline app/views/merchants/invoices/show.html.erb:25]: Nice work on the table here. Looks great!

File: app/models/item.rb [hunk 3]
```diff
@@ -26,4 +26,10 @@ def self.ready_to_ship
     .group("items.id, invoices.id")
     .order("invoice_creation asc")
   end
+
+  def self.merchant_invoices
+    joins(:invoices)
+    .select("invoices.id")
+    .distinct
+  end
```
Comment (mkrumholz) [inline app/models/item.rb:34]: Wait -- isn't this supposed to just be invoices for the merchant?? It doesn't look like we're limiting it to that here. Are you just making that happen through the view then by calling Merchant.items and only getting invoices for those items?

File: app/controllers/merchants/invoices_controller.rb (hunk 1 above)
Comment (Sidarth20) [reply app/controllers/merchants/invoices_controller.rb]: I think you're right...I can refactor this.

File: app/models/item.rb (hunk 3 above)
Comment (Sidarth20) [reply app/models/item.rb:34]: Yea, that's correct. To remove the law of demeter violation plus making sure it's only for that merchant I could just do it in the query & remove the merchant.items from the controller most likely

**Comment by mkrumholz** [comment]
Nice updates! Looks great.

**Pull request merged**